Add proper 'prev'/'next' classes to Bootstrap compact

cseelus opened the pull request:
This PR adds classes to the prev/next buttons, that enable differentiating those buttons in the DOM and to better target these buttons via CSS and JS.

cseelus committed:
Add previous/next link classes

cseelus committed:
Convert .pagy-compact-input to div to conform to HTML standards

cseelus committed:
Differentiate between prev- and next-link

ddnexus commented:
@cseelus Thanks for your pull request!

A few things:
- I would add just 'prev' and 'next' classes, as in all the other helpers. With that you should be able to select them independently (but why?)
- `pagy_link_proc` is meant to avoid calling the possibly expensive `pagy_url_to`: if you use it once per each link you produce, then it became pointless and probably counter productive. You likely don't need all that if you just want to add a couple of classes to the buttons
- can you point me to some literature about the div vs button topic that you mentioned?

Thanks

cseelus commented:
Hi @ddnexus,

thanks for the quick response.

1. Will change to just 'prev' and 'next'. One use case we implemented, is to cleanly access those links via JS to implement keyboard navigation. Also projects like [CSS Zen Garden](https://en.wikipedia.org/wiki/CSS_Zen_Garden) and practical experience thought me that its always a good idea to mark up your HTML in a way that the styling can be changed without touching the HTML later, which means using expressive class or id names that allow targeting elements individually. In our project we replaced the prev/next buttons content with Fontawesome chevron icons via CSS: <img width="389" alt="screen shot 2018-05-27 at 14 43 09" src="https://user-images.githubusercontent.com/3188392/40586072-4fda2576-61bc-11e8-98bd-759b2d11211c.png">
2. How would you go about adding those classes then?
3. The [HTML5 spec](https://www.w3.org/TR/2012/WD-html5-20120329/the-button-element.html#the-button-element) states that `<button>`s may only contain phrasing content ("Phrasing content is the text of the document, as well as elements that mark up that text at the intra-paragraph level.").

ddnexus commented:
I like the design with chevrons!

To add the classes, just remove the classes from the `pagy_link_proc` call. Then, in the `link` calls of the buttons, compose the removed classes with the next or prev class and the aria-label in one single string passed as the last argument. `link.call( page_number [, text [, extra_attributes_string ]])`

And another small thing: please, leave the tags and link assignments on the same line :)

cseelus committed:
Make prev/next class names more consistent

cseelus commented:
Thanks.

Changes are implemented. 

The tags and the link assignments in one line is to keep the line count as low as possible?

ddnexus commented:
Thanks. Actually the single line is not for line count. It's because the first line is just an initialization that I want to visually keep less important than the content below.

I don't know if that makes sense to others, but to me, the indentation style and visual disposition of statements is very suggestive of how to think about it's logic. Not to mention the bad vibes from my synesthesia :)

cseelus commented:
Ok, now it makes sense. The indentation of the code that generates the output also makes sense.

ddnexus commented:
I am going to integrate the pagy-extras gem into pagy, so this gem/repo will die very soon.

I just pushed the `dev` branch in the pagy repo with all the new changes. I would like to preserve your contribution in the new place for the compact extra, so please, could you create the same  PR (1 squashed commit is enough) based on the pagy dev branch? 

The file is at this path https://github.com/ddnexus/pagy/blob/dev/lib/pagy/extras/compact.rb. While at it, could you also add the prev/next classes to the regular `pagy_nav_compact`? Thanks.

cseelus commented:
Will do, but `pagy_nav_compact` already has prev/next classes inside those `<span>`s?

ddnexus commented:
Oh... you are right! I though I missed them everywhere! :)
